### ISSUE TITLE
Introduce a command line flag to label annotations with allele numbers instead of abbreviated sequences

### ISSUE THREAD

**Issue by stephlo76** [pcingola/SnpEff#197]
when automatically assigning annotations to different alleles, I run into problems since I do not quite understand the rules how allele sequences are abbreviated. E.g.
 
Ref: ACACG; alt: ACAC,ACACGAA -> alleles are listed as original (ACAC,ACACGAA)
Ref: ACACG; alt: ACACACGCACA,ACACA -> first is abbreviated as ACGCACA, second listed as ACACA (but both alleles contain this sequence)
Ref: TTGTGTGTGTGTGG; alt: TTGTGTGTGTGTGTGT,TTGTGTGTGTGTGTGTGT -> first is abbreviated as TGT (both contain this sequence), second is abbreviated as TGTGT(both contain this sequence)
 
Especially in the third example, the allocation of annotations to alleles is quite tricky, and I could not write a script to do this allocation automatically. Could you please introduce a command line flag so alleles are referenced by their numbers and not by their abbreviated sequence? That would make parsing the output a lot easier.

Thanks,

Stephan

**Comment by stephlo76** [comment]
Alternatively, a command line switch could disable the abbreviation of allele sequences in the annotations and output the complete allele sequences

**Comment by stephlo76** [comment]
When ref is GGGGG and alt is AGGGT,GAGGGT, *both* alleles are abbreviated as AGGGT in the annotations so one has to allocate the effects manually (in this case, one makes a frameshift and the other not):

Ld29_v01s1	776041	.	GGGGG	AGGGT,GAGGGT	67.8687	.	AB=0.4,0.5;ABP=3.44459,3.0103;AC=4,1;AF=0.666667,0.166667;AN=6;AO=4,2;CIGAR=1X3M1X,1M1I3M1X;DP=7;DPB=9.4;DPRA=0,2;EPP=11.6962,3.0103;EPPR=0;GTI=0;LEN=5,6;MEANALT=1.66667,3;MQM=37,42;MQMR=0;NS=3;NUMALT=2;ODDS=1.40895;PAIRED=1,1;PAIREDR=0;PAO=0,1;PQA=0,31.5;PQR=103.5;PRO=3;QA=148,70;QR=0;RO=0;RPL=0,0;RPP=11.6962,7.35324;RPPR=0;RPR=4,2;RUN=1,1;SAF=4,1;SAP=11.6962,3.0103;SAR=0,1;SRF=0;SRP=0;SRR=0;TYPE=complex,complex;ANN=AGGGT|frameshift_variant&missense_variant|HIGH|LdBPK_291760.1|LdBPK_291760.1|transcript|LdBPK_291760.1.1|protein_coding|2/2|c.425_428delCCCCinsACCCT|p.Pro142fs|428/7164|425/7164|142/2387||,AGGGT|missense_variant|MODERATE|LdBPK_291760.1|LdBPK_291760.1|transcript|LdBPK_291760.1.1|protein_coding|2/2|c.425_429delCCCCCinsACCCT|p.Pro142His|429/7164|425/7164|142/2387||,AGGGT|downstream_gene_variant|MODIFIER|LdBPK_291770.1|LdBPK_291770.1|transcript|LdBPK_291770.1.1|protein_coding||c.*1193_*1197delCCCCCinsACCCT|||||1197|,AGGGT|downstream_gene_variant|MODIFIER|LdBPK_291770.1|LdBPK_291770.1|transcript|LdBPK_291770.1.1|protein_coding||c.*1193_*1196delCCCCinsACCCT|||||1196|;LOF=(LdBPK_291760.1|LdBPK_291760.1|1|1,00)	GT:DP:DPR:RO:QR:AO:QA:GL	.1/2:4:4,1,2:0:0:1,2:34,70:-7.45116,-5.64438,-5.34335,-1.83744,0,-1.23538	0/1:1:1,1,0:0:0:1,0:38,0:-2.75139,0,-5.73799,-3.35345,-6.03902,-9.03233	1/1:2:2,2,0:0:0:2,0:76,0:-6.88151,-1.80773,0,-8.08563,-1.80773,-6.88151	.

**Comment by pcingola** [comment]
Hi,
  I'll try to take a look at this.
  Could you share the command line you are using and attach a VCF file with the examples you've shown? It would make testing and debugging a lot easier for me.
  Thank you!

**Comment by rasmushenningsson** [comment]
Hi!
I've run into the same problem. Below is a tiny example VCF.
As you can see, both ALTs are represented by 'T' rather than their full description in the ANN subfields. It makes it impossible to parse the annotated VCF automatically.

I'm assuming that it is a bug, since the documentation for the ALT sub-field states: "In case of multiple ALT fields, this helps to identify which ALT we are referring to."

It seems like the bug is sometimes (but not always) triggered when the REF and ALT have a common prefix.


##fileformat=VCFv4.2
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	sampleA	sampleB
chr3	33434954	.	TAAA	TAT,TT	0.0	FAIL_NVAF	SSC=0	GT:DP	0/1:44	0/0:50

which will produce the output:

##fileformat=VCFv4.2
##SnpEffVersion="4.3t (build 2017-11-24 10:18), by Pablo Cingolani"
##SnpEffCmd="SnpEff  -noStats hg19 .\minimal.vcf "
...
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	sampleA	sampleB
chr3	33434954	.	TAAA	TAT,TT	0.0	FAIL_NVAF	SSC=0;ANN=T|intron_variant|MODIFIER|UBP1|UBP1|transcript|NM_001128161.1|protein_coding|14/16|c.1391-11_1391-9delTTTinsA||||||,T|intron_variant|MODIFIER|UBP1|UBP1|transcript|NM_001128161.1|protein_coding|14/16|c.1391-11_1391-10delTTinsA||||||,T|intron_variant|MODIFIER|UBP1|UBP1|transcript|NM_001128160.1|protein_coding|12/14|c.1283-11_1283-9delTTTinsA||||||,T|intron_variant|MODIFIER|UBP1|UBP1|transcript|NM_001128160.1|protein_coding|12/14|c.1283-11_1283-10delTTinsA||||||,T|intron_variant|MODIFIER|UBP1|UBP1|transcript|NM_014517.4|protein_coding|13/15|c.1391-11_1391-9delTTTinsA||||||,T|intron_variant|MODIFIER|UBP1|UBP1|transcript|NM_014517.4|protein_coding|13/15|c.1391-11_1391-10delTTinsA||||||	GT:DP	0/1:44	0/0:50

**Comment by pcingola** [comment]
Closing old issues.

**Comment by rasmushenningsson** [comment]
I understand the need to close old issues, but please consider reopening this one. The bug is still present in 5.0, and causes a lot of problem when writing scripts/tools that interpret the snpEff output vcf, since we cannot know which ANN entry belongs to which ALT.

This is the command I ran for 1.5:
```
java -Xmx4g -jar /path/to/snpEff_1.5/snpEff.jar ann GRCh37.75 buginput.vcf > bugoutput.vcf
```
I've attached the input and output vcf here: [files.zip](https://github.com/pcingola/SnpEff/files/5084394/files.zip)
Like before, all ANN entries incorrectly get T as their ALT instead of TAT or TT, making it impossible to distinguish between them.

**Comment by stephlo76** [comment]
I would also recommend to reopen this issue. These ambiguous, obviously "abbreviated"  ALT sequences lead to a lot of problems when parsing the output. It would be great to have a switch to disable these "abbreviations" like, in the prevous examples, both  AGGGT and GAGGGT to AGGT or both TAT and TT to T since the output is very hard to parse in these cases otherwise.

**Comment by mglubber** [comment]
I'm having similar issues; it's especially problematic when working with large numbers of samples, as it makes it difficult to determine variants effect which samples. Just using the ALT number instead of the 'abbreviated' sequence would greatly simplify things, and would make the annotations easier to read.
